lib/metrics.js

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const EXPORTED_SYMBOLS = ['Metrics'];
+
+function Metrics(opts) {
+  /*
+  The metrics class is used to determine whether the recommendation is used
+  when the user navigates with the popup open. Data is sent to the Test Pilot
+  add-on, which forwards it to the Telemetry server.
+  */
+  this.events = opts.events;
+  this.interaction = null;
+  this.observerService = opts.observerService;
+  this.win = opts.win;
+  this.recommendation = opts.recommendation;
+  this.popup = opts.popup;
+  this.checked = false;
+
+  this.interactionStart = this.interactionStart.bind(this);
+  this.interactionEnd = this.interactionEnd.bind(this);
+  this.onKeyNavigate = this.onKeyNavigate.bind(this);
+  this.onClickNavigate = this.onClickNavigate.bind(this);
+}
+
+Metrics.prototype = {
+  init: function() {
```

Comment:
:beers: Yeah, unobtrusive measurement is one of the nicest things about extremely late binding via pubsub (unit testing is another). If we had modules calling each other directly, or used promise-based APIs, it'd be a lot messier.